src/colvarcomp.cpp

```diff
@@ -198,6 +196,8 @@ cvm::atom_group *colvar::cvc::parse_group(std::string const &conf,
                                COLVARS_INPUT_ERROR);
       delete group;
       group = nullptr;
+      // Silence unused variable warning; TODO stop returning a pointer
```

Comment:
Could this function become an `atom_group` constructor? That would seem the most logical to me.

Reply:
Yes, totally! However, I was also thinking about removing all those raw pointers and simply having the following as a member of the `cvc` base class:
```
std::map<std::string, std::shared_ptr<cvm::atom_groups>> groups_;
```
Then after that, it'll be easier for the CVCs to become lighter objects that operate on shared groups.

All that seems a bit out of scope and this PR is fat enough IMO.